client/snippets/snippets.json

```diff
@@ -39,5 +52,267 @@
             "$0"
         ],
         "description": "Textualization attribute"
+    },
+    "Noun Phrase": {
+        "prefix": "\\nounPhrase",
+        "body": "\\nounPhrase(\"${1:root}\", _NUMBER: ${2:number})",
+        "description": "Noun phrase"
+    },
+    "Function": {
+        "prefix": "function",
+        "body": [
+            "function ${1:func_name}(${2:args})",
+            "--> domains ${3|Boolean,Decimal,Integer,Number,Collection,Text,Object,Void|}",
+            "--> action",
+            "{",
+            "\t$0",
+            "}",
+            "; // $1"
+        ],
+        "description": "Function"
+    },
+    "Simple Function": {
+        "prefix": "function",
+        "body": [
+            "function ${1:func_name}(${2:args})",
+            "--> return",
+            "\t$0",
+            ";"
+        ],
+        "description": "Function with direct return"
+    },
+    "Text Function": {
+        "prefix": "TextFunction",
+        "body": [
+            "TextFunction ${1:func_name}(${2:args})",
+            "--> text",
+            "\t\\(",
+            "\t\t$0",
+            "\t\\)",
+            "; // $1"
+        ],
+        "description": "Text function"
+    },
+    "Verb": {
+        "prefix": "\\verb",
+        "body": "\\verb(VERB_${1|EN,FR,DE,ES,JA,NL,PT|}_${2:VERBNAME}, ${3|PRESENT_INDICATIVE,AUX_MOD,AUX_MOD_CONTINUOUS,AUX_MOD_PAST_PERFECT,AUX_MOD_PAST_PERFECT_CONTINUOUS,AUX_MOD_PERFECT,AUX_MOD_PERFECT_CONTINUOUS,AUX_MOD_PRETERIT,AUX_MOD_PRETERIT_CONTINUOUS,CONDITIONAL,CONJUNCTIVE_ROOT,CONTINUOUS_FUTURE,CONTINUOUS_FUTURE_PERFECT,CONTINUOUS_PAST_CONDITIONAL,CONTINUOUS_PAST_PERFECT,CONTINUOUS_PRESENT,CONTINUOUS_PRESENT_CONDITIONAL,CONTINUOUS_PRESENT_PERFECT,CONTINUOUS_PRETERIT,FUTURE,FUTURE_1_INDICATIVE,FUTURE_2_INDICATIVE,FUTURE_INDICATIVE,FUTURE_PERFECT,FUTURE_PERFECT_INDICATIVE,FUTURE_PERFECT_SUBJUNCTIVE,FUTURE_SUBJUNCTIVE,FUTURE_SUBJUNCTIVE_1,FUTURE_SUBJUNCTIVE_2,GERUNDIVE,IMPERATIVE,IMPERFECT_INDICATIVE,IMPERFECT_SUBJUNCTIVE,INACCOMPLISHED_ROOT,INFINITIVE,INFINITIVE_1,INFINITIVE_2,PARTICIPLE,PARTICIPLE_1,PARTICIPLE_2,PARTICIPLE_3,PAST,PAST_CONDITIONAL,PAST_CONDITIONAL_SECOND_FORM,PAST_GERUNDIVE,PAST_HISTORIC_INDICATIVE,PAST_IMPERATIVE,PAST_INFINITIVE,PAST_PARTICIPLE,PAST_PERFECT,PAST_PERFECT_INDICATIVE,PAST_SUBJUNCTIVE,PAST_SUBJUNCTIVE_1,PAST_SUBJUNCTIVE_2,PERFECT_INDICATIVE,PLU_PERFECT_INDICATIVE,PLU_PERFECT_SUBJUNCTIVE,POTENTIAL,PRESENT,PRESENT_CONDITIONAL,PRESENT_IMPERATIVE,PRESENT_INFINITIVE,PRESENT_OR_FUTURE,PRESENT_PARTICIPLE,PRESENT_PERFECT,PRESENT_PERFECT_INDICATIVE,PRESENT_SUBJONCTIVE_1,PRESENT_SUBJONCTIVE_2,PRESENT_SUBJUNCTIVE,PRETERIT,PRETERIT_INDICATIVE,TE_FORM,VOLITIONAL|}_$1, _SUBJECT_NUMBER: ${4:nb})",
```

Comment:
* We talked about this before. Since I found to way to make the second list depend on the previous choices, I had to choose between:
    1. Include the language code in the choices. Disadvantage: if you start to type “present”, you might get four or five suggestions, most of them being for irrelevant languages.
    2. Do as I did here. Disadvantage: you kinda have to know what you're looking for (but then you get more pertinent suggestions and can spam `Enter`).
* “they don't know how they are named” → Wut? How have they been writing their code without suggestions at all, then? And I don't think this snippet will make this phenomenon worse, anyway, regardless of the chosen implementation (see two-item list above). Furthermore, future CLI tools should help quickly detect tense / language mismatches and such.